test/Core.Test/AdminConsole/OrganizationAuth/UpdateOrganizationAuthRequestCommandTests.cs

```diff
@@ -76,25 +78,143 @@ await sutProvider.GetDependency<IMailService>().DidNotReceive()
+            Arg.Any<string>(),
+            Arg.Any<string>()
+        );
+        await sutProvider.GetDependency<IEventService>().ReceivedWithAnyArgs().LogOrganizationUserEventsAsync(
+            Arg.Any<IEnumerable<(OrganizationUser, EventType, DateTime?)>>()
+        );
+    }
+
+    [Theory]
+    [BitAutoData]
+    public async Task UpdateAsync_BatchUpdate_AuthRequestIsDenied_DoesNotLeakRejection(
+        SutProvider<UpdateOrganizationAuthRequestCommand> sutProvider,
+        List<OrganizationAuthRequestUpdate> updates,
+        OrganizationAdminAuthRequest unprocessedAuthRequest,
+        AuthRequestUpdateProcessorConfiguration configuration,
+        User user
+    )
+    {
+        // For this command to work we need the following from external
+        // classes:
+        // 1. A configured expiration timespan for organization auth requests
+        // 2. Some unresponded to auth requests that match the ids provided
+        // 3. A valid user to send emails to
+
+        var unprocessedAuthRequests = new List<OrganizationAdminAuthRequest>();
+        unprocessedAuthRequest = UnrespondAndEnsureValid(unprocessedAuthRequest, configuration.OrganizationId);
+        foreach (var update in updates)
+        {
+            update.Approved = false;
+            unprocessedAuthRequest.Id = update.Id;
+            unprocessedAuthRequests.Add(unprocessedAuthRequest);
+        };
+
+        sutProvider.GetDependency<IGlobalSettings>().PasswordlessAuth.AdminRequestExpiration.Returns(TimeSpan.FromDays(7));
+
+        sutProvider.GetDependency<IAuthRequestRepository>().GetManyAdminApprovalRequestsByManyIdsAsync(
+            configuration.OrganizationId,
+            updates.Select(ar => ar.Id)
+        ).ReturnsForAnyArgs(unprocessedAuthRequests);
+
+        sutProvider.GetDependency<IUserRepository>().GetByIdAsync(Arg.Any<Guid>()).Returns(user);
+
+        // Call the SUT
+        await sutProvider.Sut.UpdateAsync(configuration.OrganizationId, updates);
+
+        // Assert that because we passed in good data we call a save
+        // operation and raise all events
+        await sutProvider.GetDependency<IAuthRequestRepository>().ReceivedWithAnyArgs().UpdateManyAsync(Arg.Any<IEnumerable<OrganizationAdminAuthRequest>>());
+        await sutProvider.GetDependency<IPushNotificationService>().DidNotReceiveWithAnyArgs().PushAuthRequestResponseAsync(Arg.Any<AuthRequest>());
+        await sutProvider.GetDependency<IMailService>().DidNotReceiveWithAnyArgs().SendTrustedDeviceAdminApprovalEmailAsync(
+            Arg.Any<string>(),
+            Arg.Any<DateTime>(),
+            Arg.Any<string>(),
+            Arg.Any<string>()
+        );
+        await sutProvider.GetDependency<IEventService>().ReceivedWithAnyArgs().LogOrganizationUserEventsAsync(
+            Arg.Any<IEnumerable<(OrganizationUser, EventType, DateTime?)>>()
+        );
+    }
+
+    private T UnrespondAndEnsureValid<T>(T authRequest, Guid organizationId) where T : AuthRequest
```

Comment:
Non-blocking, for next time only: You can use Customization attributes to help you arrange the autogenerated data. e.g. https://github.com/bitwarden/server/blob/b84c31225757f07b971572acbb23799c47b45df6/test/Core.Test/AdminConsole/AutoFixture/OrganizationFixtures.cs#L1-L1

An alternative is not to use `BitAutoData` and just define a factory function that gives you the fixture that you want, which is sometimes easier than trying to figure out how to arrange the autodata.

There's nothing wrong with this as well, just throwing around ideas for next time.